metron-analytics/metron-profiler-client/README.md

```diff
@@ -91,37 +60,268 @@ want to change the global Client configuration so as not to disrupt the work of
+   * <span style="color:blue">`from 1 hour`</span>
+   * <span style="color:blue">`starting from 1 hour ago`</span>
+   * <span style="color:blue">`starting from 1 hour`</span>
+* A duration starting 1 hour ago and ending 30 minutes ago: 
+   * <span style="color:blue">`from 1 hour ago until 30 minutes ago`</span>
+   * <span style="color:blue">`from 30 minutes ago until 1 hour ago`</span>
+   * <span style="color:blue">`starting from 1 hour ago to 30 minutes ago`</span>
+   * <span style="color:blue">`starting from 1 hour to 30 minutes`</span>
+
+###### <span style="color:red">Temporal Window Width</span>
+
+Temporal window width is the specification of a window. 
+A window is may either repeat within total temporal duration or may fill the total temporal duration.
+A window is specified by the phrase: `time_interval WINDOW`
+* `time_interval` - A time amount followed by a unit (e.g. 1 hour).  The unit may be "minute", "day", "hour" with any pluralization.
+* `WINDOW` - Optionally the word "window"
+
+**Examples**
+
+* A fixed window starting 2 hours ago and going until now
+  * <span style="color:red">`2 hour`</span>
+  * <span style="color:red">`2 hours`</span>
+  * <span style="color:red">`2 hours window`</span>
+* A repeating 30 minute window starting 2 hours ago and repeating every hour until now.
+This would result in 2 30-minute wide windows: 2 hours ago and 1 hour ago
+  * <span style="color:red">`30 minute window`</span><span style="color:green">`every 1 hour`</span><span style="color:blue">`starting from 2 hours ago`</span>
+  * <span style="color:red">`30 minutes window`</span><span style="color:green">`every 1 hour`</span><span style="color:blue">`from 2 hours ago`</span>
+* A repeating 30 minute window starting 2 hours ago and repeating every hour until 30 minutes ago.
+This would result in 2 30-minute wide windows: 2 hours ago and 1 hour ago
+  * <span style="color:red">`30 minute window`</span><span style="color:green">`every 1 hour`</span><span style="color:blue">`starting from 2 hours ago until 30 minutes ago`</span>
+  * <span style="color:red">`30 minutes window`</span><span style="color:green">`every 1 hour`</span><span style="color:blue">`from 2 hours ago to 30 minutes ago`</span> 
+  * <span style="color:red">`30 minutes window`</span><span style="color:green">`for every 1 hour`</span><span style="color:blue">`from 30 minutes ago to 2 hours ago`</span> 
+
+###### <span style="color:green">Skip distance</span>
+
+Skip distance is the amount of time between temporal window beginnings that the next window starts.
+It is, in effect, the window period.  
+
+It is specified by the phrase `EVERY time_interval`
+* `time_interval` - A time amount followed by a unit (e.g. 1 hour).  The unit may be "minute", "day", "hour" with any pluralization.
+* `EVERY` - The word/phrase "every" or "for every"
+
+**Examples**
+
+* A repeating 30 minute window starting 2 hours ago and repeating every hour until now.
```

Comment:
How do I select the last 1 hour window for the past 8 Tuesdays (assuming today is Tuesday)?

How do I select the last 1 hour window for the past 8 'current day of the week'?